### PR TITLE
ci: Release Fury Java Snapshot 

### PR DESCRIPTION
**Pull request by tisonkun**

### CODE REVIEW

**Review by chaokunyang — APPROVED** [review]
LGTM

File: java/pom.xml [hunk 1]
```diff
@@ -261,35 +254,4 @@
       </plugin>
     </plugins>
   </build>
-
-  <profiles>
-    <profile>
-      <id>release</id>
```
Comment (chaokunyang) [inline java/pom.xml:267]: Just found that remove this part will skip jar release in sub modules.

Fury use properties to skip module deploy, deploy plugin won't be enabled without this profile

Comment (tisonkun) [reply java/pom.xml:267]: No. They are released - https://repository.apache.org/content/groups/snapshots/org/apache/fury/

We now inherit with Apache Maven Parent POM which contains an` apache-release` profile to help.

Comment (chaokunyang) [reply java/pom.xml:267]: I see, thanks

**Pull request merged**